public/app/percona/check/components/ServiceChecks/ServiceChecks.tsx

```diff
@@ -0,0 +1,183 @@
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import PageWrapper from 'app/percona/shared/components/PageWrapper/PageWrapper';
+import { Table } from 'app/percona/integrated-alerting/components/Table';
+import { useCancelToken } from 'app/percona/shared/components/hooks/cancelToken.hook';
+import { useStoredTablePageSize } from 'app/percona/integrated-alerting/components/Table/Pagination';
+import { CheckService } from 'app/percona/check/Check.service';
+import { isApiCancelError } from 'app/percona/shared/helpers/api';
+import { ServiceFailedCheck } from '../../types';
+import { PAGE_MODEL } from '../../CheckPanel.constants';
+import { SERVICE_CHECKS_CANCEL_TOKEN, SERVICE_CHECKS_TABLE_ID } from './ServiceChecks.constants';
+import { Messages } from './ServiceChecks.messages';
+import { getStyles } from './ServiceChecks.styles';
+import { Severity } from 'app/percona/integrated-alerting/components/Severity';
+import { formatServiceId } from '../FailedChecksTab/FailedChecksTab.utils';
+import { SilenceBell } from 'app/percona/shared/components/Elements/SilenceBell';
+import { ExpandableCell } from 'app/percona/shared/components/Elements/ExpandableCell';
+
+export const ServiceChecks: FC<GrafanaRouteComponentProps<{ service: string }>> = ({ match }) => {
+  const serviceId = formatServiceId(match.params.service);
+  const [pageSize, setPageSize] = useStoredTablePageSize(SERVICE_CHECKS_TABLE_ID);
+  const [pageIndex, setPageindex] = useState(0);
+  const [totalItems, setTotalItems] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+  const [data, setData] = useState<ServiceFailedCheck[]>([]);
+  const [pending, setPending] = useState(false);
+  const [serviceName, setServiceName] = useState('');
+  const [generateToken] = useCancelToken();
+  const styles = useStyles2(getStyles);
+
+  const fetchChecks = useCallback(async () => {
+    try {
+      setPending(true);
+      const {
+        data,
+        totals: { totalItems, totalPages },
+      } = await CheckService.getFailedCheckForService(
+        serviceId,
+        pageSize,
+        pageIndex,
+        generateToken(SERVICE_CHECKS_CANCEL_TOKEN)
+      );
+      setData(data);
+      setServiceName(data[0].serviceName);
+      setTotalItems(totalItems);
+      setTotalPages(totalPages);
+    } catch (e) {
+      if (isApiCancelError(e)) {
+        return;
+      }
+      logger.error(e);
+    }
+    setPending(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pageIndex, pageSize, serviceId]);
+
+  const onSilenceClick = useCallback(
+    async (alertId: string, silenced: boolean) => {
+      await CheckService.silenceAlert(alertId, !silenced);
+      fetchChecks();
+    },
+    [fetchChecks]
+  );
+
+  const columns = useMemo(
+    (): Array<Column<ServiceFailedCheck>> => [
+      {
+        Header: 'Check Name',
+        accessor: 'checkName',
+        // eslint-disable-next-line react/display-name
```

Comment:
Maybe this can be disabled for the entire file: https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments

Reply:
👍